scripts/setup/bootstrap.sh

```diff
@@ -167,8 +167,14 @@ EOF
 # bootstrap or run_in_build_env.sh can be executed in a build env
 _ORIGINAL_PW_ENVIRONMENT_ROOT="$PW_ENVIRONMENT_ROOT"
 
+# pigweed does not seem to handle pwd involving symlinks very well.
+current_dir=$(pwd)
+cd $(realpath $current_dir)
+
 _bootstrap_or_activate "$0"
 
+cd $current_dir
```

Comment:
```suggestion
# Workaround for https://github.com/project-chip/connectedhomeip/issues/31851
if hash realpath 2>/dev/null; then
    realpwd="$(realpath "$PWD")"
    if [ "$realpwd" != "$PWD" ]; then
        echo "Warning: $PWD contains symlinks, using $realpwd instead"
        cd "$realpwd"
    fi
fi

_bootstrap_or_activate "$0"
```

If realpath is not available we can just carry on, so we don't have to install extra stuff in CI

Reply:
(Ideally the comment should reference the pigweed issue, once one has been filed)

Reply:
Ah, that is a good point.

Reply:
Made this change.